core/src/main/java/org/pentaho/platform/workitem/WorkItemLifecyclePublisher.java

```diff
@@ -0,0 +1,135 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2017 Pentaho Corporation. All rights reserved.
+ */
+
+package org.pentaho.platform.workitem;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.platform.engine.core.system.PentahoSessionHolder;
+import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Date;
+
+/**
+ * A class for common utility methods related to work item lifecycles.
+ */
+@Configuration
+public class WorkItemLifecyclePublisher {
+
+  private static String PUBLISHER_BEAN_NAME = "workItemLifecyclePublisher";
+  private static final Log log = LogFactory.getLog( WorkItemLifecyclePublisher.class );
+
+  @Autowired
+  private ApplicationEventPublisher publisher = null;
+
+  public void setApplicationEventPublisher( final ApplicationEventPublisher publisher ) {
+    this.publisher = publisher;
+  }
+
+  public ApplicationEventPublisher getApplicationEventPublisher() {
+    return this.publisher;
+  }
+
+  /**
+   * A convenience method for publishing changes to the work item's lifecycles that calls
+   * {@link #publish(String, String, WorkItemLifecyclePhase, String, Date)} with a null {@code lifecycleDetails} and
+   * {@code sourceTimestamp}
+   *
+   * @param workItemUid            a {@link String} containing unique identifier for the {@link WorkItemLifecycleEvent}
+   * @param workItemDetails        a {@link String} containing details of the {@link WorkItemLifecycleEvent}
+   * @param workItemLifecyclePhase a {@link WorkItemLifecyclePhase} representing the lifecycle event
+   */
+  public static void publish( final String workItemUid, final String workItemDetails, final WorkItemLifecyclePhase
```

Comment:
Should workItemdetails be a bean ? Hard to know what to pass in when its a string.